### PR TITLE
Cart Management , WishList , Added Auth and API's

### PR DESCRIPTION
**Pull request by ashishkatiyar2404**
Screenshot (1021)
Screenshot (1022)

### CODE REVIEW

**Review by shashikantmaurya0007 — APPROVED** [review]
code looks good to be merged

**Review by varun1233 — APPROVED** [review]
Code looks good to merge. Just remove consoles and commented code if not required

File: music-ecom/src/Components/Navbar/Navbar.js [hunk 1]
```diff
@@ -1,32 +1,72 @@
 import React from "react";
 import "./Navbar.css";
 import { FaSearch, FaHeart, FaShoppingCart } from "react-icons/fa";
-import { Link } from "react-router-dom";
+import { Link, useNavigate } from "react-router-dom";
+import { useWishList } from "../../Context/WishListContext";
+import { useCart } from "../../Context/CartContext";
+import { useAuth } from "../../Context/AuthContext";
```
Comment (ShrutiLakhani) [inline music-ecom/src/Components/Navbar/Navbar.js:7]: You may create a separate file as context and import everything from there

File: music-ecom/src/Context/AuthContext.js [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    AuthInitialState
+  );
+  useEffect(() => {
+    function oneTime(AuthDispatch) {
+      console.log("I am here in USEEFFECT");
+      if (localStorage.getItem("loginToken" !== null)) {
+        console.log("Login is continuing");
+        return AuthDispatch({ type: "LOGIN_PERSIST" });
+      } else {
+        console.log("Masla hgya ji USEEFFECT");
```
Comment (ShrutiLakhani) [inline music-ecom/src/Context/AuthContext.js:23]: Interesting console log messages .. But better to toastify it with sensible alerts :)

**Review by ShrutiLakhani — APPROVED** [review]
Code looks good. Suggested few changes, you may want to have a look. Good work on features 👍

File: music-ecom/src/Components/Navbar/Navbar.css [hunk 3]
```diff
@@ -27,6 +33,16 @@
   font-size: 1.2rem;
 }
 
+.dynamic__btn button {
+  border: 0;
+  border-radius: 1rem;
+  height: 2.2rem;
+  background-color: transparent;
+  color: #1e6d43;
```
[the remark this answers is not in the document]
Comment (ashishkatiyar2404) [reply music-ecom/src/Components/Navbar/Navbar.css:41]: okay

File: music-ecom/src/Components/Navbar/Navbar.js (hunk 1 above)
Comment (ashishkatiyar2404) [reply music-ecom/src/Components/Navbar/Navbar.js:7]: Will do that.

File: music-ecom/src/Components/ProductCard/ProductCard.js [hunk 4]
```diff
@@ -15,25 +32,52 @@ export default function ProductCard({ items }) {
...
+              <button
+                // className="add_to_wishlist"
+                // disabled={stored}
+                onClick={() =>
+                  wishDispatch({ type: "ADD_TO_WISHLIST", payload: items })
+                }
+              >
+                Add to WishList
+              </button>
+            )}
```
[the remark this answers is not in the document]
Comment (ashishkatiyar2404) [reply music-ecom/src/Components/ProductCard/ProductCard.js:80]: okay

File: music-ecom/src/Context/AuthContext.js (hunk 2 above)
Comment (ashishkatiyar2404) [reply music-ecom/src/Context/AuthContext.js:23]: ok😅

**Pull request merged**